app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -22,20 +19,24 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     api :GET, "/organizations/:organization_id/subscriptions", N_("List organization subscriptions")
     api :GET, "/systems/:system_id/subscriptions", N_("List a system's subscriptions"), :deprecated => true
     api :GET, "/activation_keys/:activation_key_id/subscriptions", N_("List an activation key's subscriptions")
+    api :GET, "/subscriptions"
+    param_group :search, Api::V2::ApiController
     param :organization_id, :number, :desc => N_("Organization ID"), :required => true
     param :system_id, String, :desc => N_("UUID of the system"), :required => false
     param :activation_key_id, String, :desc => N_("Activation key ID"), :required => false
-    param_group :search, Api::V2::ApiController
     def index
-      subscriptions = if @system
-                        index_system
-                      elsif @activation_key
-                        index_activation_key
-                      else
-                        index_organization
-                      end
+      respond(:collection => scoped_search(index_relation.uniq, :cp_id, :asc, :resource_class => Pool))
+    end
 
-      respond_for_index(:collection => subscriptions)
+    def index_relation
+      collection = Pool.scoped
```

Comment:
That is a fair assessment, that is the way it has been, to change it to better reflect the candlepin database would require a complete overhaul of this part of the code (as you had suggested), and changes to the API, but I think that is a different project than just converting subscriptions to scoped search?